lib/table.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'row'
+
+class Table
+  def initialize(node)
+    @table = node
+  end
+
+  def to_a
```

Comment:
This is quite a complicated method. Parts of this should almost certainly be split out into individual methods. I think it would probably also much clearer if it followed the same pattern of using field_serializer rather than taking it own different approach.